demo-apps/cf-android/app/src/main/java/org/eclipse/californium/examples/MainActivity.java

```diff
@@ -89,6 +107,15 @@ protected void onPreExecute() {
 
         protected CoapResponse doInBackground(String... args) {
             CoapClient client = new CoapClient(args[0]);
+            if (isSecure) {
+                CoapEndpoint.CoapEndpointBuilder endpointBuilder = new CoapEndpoint.CoapEndpointBuilder();
+                DtlsConnectorConfig.Builder dtlsConfig = new DtlsConnectorConfig.Builder();
+                dtlsConfig.setClientOnly();
+                ConfigureDtls.loadCredentials(getApplicationContext(), dtlsConfig, CLIENT_NAME);
+                DTLSConnector dtlsConnector = new DTLSConnector(dtlsConfig.build());
+                endpointBuilder.setConnector(dtlsConnector);
+                client.setEndpoint(endpointBuilder.build());
```

Comment:
With that, every request creates it's own dtls connector, always trigger a new handshake and leave to old one running on the devices RAM. Is that intended? If not consider to use

`        EndpointManager.getEndpointManager().setDefaultEndpoint( ... );`

Please ensure, that you test your App also for several request, and also with a request after a longer pause.
If you intend to use it with an other server then "localhost", or to use you App also "on the way", please also check, if it works, when changing your location (sometime you may get a new ip address, if you move with your device). If it seems to stuck and the request are reported to timeout,  then you may require a "DTLS reset session" Menu or Button and/or configure "AutoResumptionTimeoutMillis" (see also issue #508).

If it's just intended to run for localhost, I would appreciate, if that is mentioned.

Reply:
creating new dtls connector on every request is a mistake, I will correct it now. For now I would add just local dtls server, in future commits I will add the feature of using with other dtls server as well.
I will add session auto resumption after few millis, and as well a button to stop the session.